### PR TITLE
Add basic test for the plugin

### LINKED ISSUE #6: No tests / Tests missing
Hi,
  this plugin has no test at all, this is a situation that should be amended soon, so for example changes in both the plugin or the logstash core himself, are properly tested in the CI environment.

If you are giving this plugin some test, feel free to update https://github.com/elastic/logstash/issues/3740 so we keep track of it.

### PR DESCRIPTION
**Pull request by purbon**
Fixes #6

### CODE REVIEW

**Comment by purbon** [comment]
@colinsurprenant would be nice if you could take a look here and let me know what do you think. Thanks a lot for your time.

File: logstash-output-gelf.gemspec [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Gem::Specification.new do |s|
 
   s.name            = 'logstash-output-gelf'
-  s.version         = '1.0.0'
+  s.version         = '1.1.0'
```
Comment (ph) [inline logstash-output-gelf.gemspec:4]: 1.0.1 instead of 1.1.0, we did not change the interface.

File: lib/logstash/outputs/gelf.rb [hunk 2]
```diff
@@ -81,12 +81,22 @@ class LogStash::Outputs::Gelf < LogStash::Outputs::Base
...
+    @gelf
+  end
+
   def register
     require "gelf" # rubygem 'gelf'
     option_hash = Hash.new
 
     #@gelf = GELF::Notifier.new(@host, @port, @chunksize, option_hash)
-    @gelf = GELF::Notifier.new(@host, @port, @chunksize)
+    @gelf ||= GELF::Notifier.new(@host, @port, @chunksize)
```
Comment (ph) [inline lib/logstash/outputs/gelf.rb:99]: It's not necessary to create an `inject_client` method in that context, what is evil is our usage or `ivar` in the plugins.
You create a reader method specifically for the gelf client.

``` ruby
def gelf
  @gelf ||= if the bundle of jars is done in the `rake publish` it should work out of the box with jarvis 
end
```

And in the test you can mock it.

```
output = Logstash::Output::Gelf.new
expect(output).to receive(:client).and_return(mocked_gelf)
```

File: spec/outputs/gelf_spec.rb [hunk 3]
```diff
@@ -1 +1,36 @@
-require "logstash/devutils/rspec/spec_helper"
+# encoding: utf-8
+require_relative "../spec_helper"
+require "gelf"
+
+describe LogStash::Outputs::Gelf do
+
+  let(:host) { "localhost" }
+  let(:port) { rand(2048)+1024 }
```
Comment (ph) [inline spec/outputs/gelf_spec.rb]: `#rand` support a range option.

```
rand(1024..65535)
```

File: spec/outputs/gelf_spec.rb [hunk 4]
```diff
@@ -1 +1,36 @@
...
+    expect { plugin.register }.to_not raise_error
+  end
+
+  describe "#send" do
+
+    subject { LogStash::Outputs::Gelf.new("host" => host, "port" => port ) }
+
+    let(:properties) { { "message" => "This is a message!"} }
+    let(:event)      { LogStash::Event.new(properties) }
+
```
Comment (ph) [inline spec/outputs/gelf_spec.rb]: space?

**Comment by ph** [comment]
Added a few comments.

File: lib/logstash/outputs/gelf.rb (hunk 2 above)
Comment (purbon) [inline lib/logstash/outputs/gelf.rb:99]: Well the idea of having an inject client is more clean in my perspective, it lets you do the same thing (dependency injection) been explicit.

**Comment by ph** [comment]
@purbon Not a big fan of the inject_client, you have addressed my concerns you can merge this.

**Pull request closed without merging**